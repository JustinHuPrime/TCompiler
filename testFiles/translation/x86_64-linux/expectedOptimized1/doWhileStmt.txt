testFiles/translation/x86_64-linux/input/doWhileStmt.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(5,
    MOVE(TEMP(temp6, 8, 8, GP), REG(rdi, 8)),
    JUMP(LOCAL(8)),
  ),
  BLOCK(8,
    U2F(TEMP(temp9, 4, 4, FP), CONSTANT(1, BYTE(1))),
    MOVE(TEMP(temp10, 4, 4, FP), TEMP(temp9, 4, 4, FP)),
    JUMP(LOCAL(14)),
  ),
  BLOCK(7,
    JUMP(LOCAL(14)),
  ),
  BLOCK(14,
    MOVE(TEMP(temp15, 4, 4, FP), TEMP(temp10, 4, 4, FP)),
    U2F(TEMP(temp16, 4, 4, FP), CONSTANT(1, BYTE(2))),
    FMUL(TEMP(temp17, 4, 4, FP), TEMP(temp15, 4, 4, FP), TEMP(temp16, 4, 4, FP)),
    MOVE(TEMP(temp10, 4, 4, FP), TEMP(temp17, 4, 4, FP)),
    JUMP(LOCAL(19)),
  ),
  BLOCK(19,
    MOVE(TEMP(temp20, 8, 8, GP), TEMP(temp6, 8, 8, GP)),
    SUB(TEMP(temp21, 8, 8, GP), TEMP(temp20, 8, 8, GP), CONSTANT(8, LONG(1))),
    MOVE(TEMP(temp6, 8, 8, GP), TEMP(temp21, 8, 8, GP)),
    JUMP(LOCAL(22)),
  ),
  BLOCK(22,
    ZX(TEMP(temp23, 8, 8, GP), CONSTANT(1, BYTE(0))),
    J2A(LOCAL(7), LOCAL(11), TEMP(temp21, 8, 8, GP), TEMP(temp23, 8, 8, GP)),
  ),
  BLOCK(11,
    MOVE(TEMP(temp25, 4, 4, FP), TEMP(temp10, 4, 4, FP)),
    JUMP(LOCAL(24)),
  ),
  BLOCK(24,
    MOVE(TEMP(temp2, 4, 4, FP), TEMP(temp25, 4, 4, FP)),
    JUMP(LOCAL(4)),
  ),
  BLOCK(4,
    MOVE(REG(xmm0, 4), TEMP(temp2, 4, 4, FP)),
    RETURN(),
  ),
)
