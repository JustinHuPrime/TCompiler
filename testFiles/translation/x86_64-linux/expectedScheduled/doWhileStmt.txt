testFiles/translation/x86_64-linux/input/doWhileStmt.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(0,
    LABEL(LOCAL(5)),
    MOVE(TEMP(temp6, 8, 8, GP), REG(rdi, 8)),
    LABEL(LOCAL(8)),
    U2F(TEMP(temp9, 4, 4, FP), CONSTANT(1, BYTE(1))),
    MOVE(TEMP(temp10, 4, 4, FP), TEMP(temp9, 4, 4, FP)),
    LABEL(LOCAL(14)),
    MOVE(TEMP(temp15, 4, 4, FP), TEMP(temp10, 4, 4, FP)),
    U2F(TEMP(temp16, 4, 4, FP), CONSTANT(1, BYTE(2))),
    FMUL(TEMP(temp17, 4, 4, FP), TEMP(temp15, 4, 4, FP), TEMP(temp16, 4, 4, FP)),
    MOVE(TEMP(temp10, 4, 4, FP), TEMP(temp17, 4, 4, FP)),
    LABEL(LOCAL(19)),
    MOVE(TEMP(temp20, 8, 8, GP), TEMP(temp6, 8, 8, GP)),
    SUB(TEMP(temp21, 8, 8, GP), TEMP(temp20, 8, 8, GP), CONSTANT(8, LONG(1))),
    MOVE(TEMP(temp6, 8, 8, GP), TEMP(temp21, 8, 8, GP)),
    LABEL(LOCAL(22)),
    ZX(TEMP(temp23, 8, 8, GP), CONSTANT(1, BYTE(0))),
    J1A(LOCAL(7), TEMP(temp21, 8, 8, GP), TEMP(temp23, 8, 8, GP)),
    LABEL(LOCAL(11)),
    MOVE(TEMP(temp25, 4, 4, FP), TEMP(temp10, 4, 4, FP)),
    LABEL(LOCAL(24)),
    MOVE(TEMP(temp2, 4, 4, FP), TEMP(temp25, 4, 4, FP)),
    LABEL(LOCAL(4)),
    MOVE(REG(xmm0, 4), TEMP(temp2, 4, 4, FP)),
    RETURN(),
    LABEL(LOCAL(7)),
  ),
)
