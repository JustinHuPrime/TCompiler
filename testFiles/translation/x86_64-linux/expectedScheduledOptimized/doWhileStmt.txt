testFiles/translation/x86_64-linux/input/doWhileStmt.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(0,
    MOVE(TEMP(temp6, 8, 8, GP), REG(rdi, 8)),
    U2F(TEMP(temp9, 4, 4, FP), CONSTANT(1, BYTE(1))),
    MOVE(TEMP(temp10, 4, 4, FP), TEMP(temp9, 4, 4, FP)),
    MOVE(TEMP(temp15, 4, 4, FP), TEMP(temp10, 4, 4, FP)),
    U2F(TEMP(temp16, 4, 4, FP), CONSTANT(1, BYTE(2))),
    FMUL(TEMP(temp17, 4, 4, FP), TEMP(temp15, 4, 4, FP), TEMP(temp16, 4, 4, FP)),
    MOVE(TEMP(temp10, 4, 4, FP), TEMP(temp17, 4, 4, FP)),
    MOVE(TEMP(temp20, 8, 8, GP), TEMP(temp6, 8, 8, GP)),
    SUB(TEMP(temp21, 8, 8, GP), TEMP(temp20, 8, 8, GP), CONSTANT(8, LONG(1))),
    MOVE(TEMP(temp6, 8, 8, GP), TEMP(temp21, 8, 8, GP)),
    ZX(TEMP(temp23, 8, 8, GP), CONSTANT(1, BYTE(0))),
    J1A(CONSTANT(8, LOCAL(7)), TEMP(temp21, 8, 8, GP), TEMP(temp23, 8, 8, GP)),
    MOVE(TEMP(temp25, 4, 4, FP), TEMP(temp10, 4, 4, FP)),
    MOVE(TEMP(temp2, 4, 4, FP), TEMP(temp25, 4, 4, FP)),
    MOVE(REG(xmm0, 4), TEMP(temp2, 4, 4, FP)),
    RETURN(),
    LABEL(CONSTANT(8, LOCAL(7))),
  ),
)
