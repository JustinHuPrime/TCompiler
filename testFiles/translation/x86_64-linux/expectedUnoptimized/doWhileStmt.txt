testFiles/translation/x86_64-linux/input/doWhileStmt.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(5,
    MOVE(TEMP(temp6, 8, 8, GP), REG(rdi, 8)),
    JUMP(LOCAL(3)),
  ),
  BLOCK(3,
    JUMP(LOCAL(8)),
  ),
  BLOCK(8,
    UNSIGNED2FLOATING(TEMP(temp9, 4, 4, FP), CONSTANT(1, BYTE(1))),
    MOVE(TEMP(temp10, 4, 4, FP), TEMP(temp9, 4, 4, FP)),
    JUMP(LOCAL(7)),
  ),
  BLOCK(7,
    JUMP(LOCAL(13)),
  ),
  BLOCK(13,
    JUMP(LOCAL(14)),
  ),
  BLOCK(14,
    MOVE(TEMP(temp15, 4, 4, FP), TEMP(temp10, 4, 4, FP)),
    UNSIGNED2FLOATING(TEMP(temp16, 4, 4, FP), CONSTANT(1, BYTE(2))),
    FMUL(TEMP(temp17, 4, 4, FP), TEMP(temp15, 4, 4, FP), TEMP(temp16, 4, 4, FP)),
    MOVE(TEMP(temp10, 4, 4, FP), TEMP(temp17, 4, 4, FP)),
    JUMP(LOCAL(12)),
  ),
  BLOCK(12,
    JUMP(LOCAL(19)),
  ),
  BLOCK(19,
    MOVE(TEMP(temp20, 8, 8, GP), TEMP(temp6, 8, 8, GP)),
    SUB(TEMP(temp21, 8, 8, GP), TEMP(temp20, 8, 8, GP), CONSTANT(8, LONG(1))),
    MOVE(TEMP(temp6, 8, 8, GP), TEMP(temp21, 8, 8, GP)),
    JUMP(LOCAL(18)),
  ),
  BLOCK(18,
    JUMP(LOCAL(22)),
  ),
  BLOCK(22,
    ZX(TEMP(temp23, 8, 8, GP), CONSTANT(1, BYTE(0))),
    J2A(LOCAL(7), LOCAL(11), TEMP(temp21, 8, 8, GP), TEMP(temp23, 8, 8, GP)),
  ),
  BLOCK(11,
    MOVE(TEMP(temp25, 4, 4, FP), TEMP(temp10, 4, 4, FP)),
    JUMP(LOCAL(24)),
  ),
  BLOCK(24,
    MOVE(TEMP(temp2, 4, 4, FP), TEMP(temp25, 4, 4, FP)),
    JUMP(LOCAL(4)),
  ),
  BLOCK(4,
    MOVE(REG(xmm0, 4), TEMP(temp2, 4, 4, FP)),
    RETURN(),
  ),
)
