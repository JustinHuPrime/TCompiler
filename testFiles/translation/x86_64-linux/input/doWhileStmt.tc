module foo;

float bar(ulong ntimes) {
  float f = 1;
  do {
    f *= 2;
  // yes, I know this isn't correct - but it should compile
  } while (--ntimes > 0)
  return f;
}