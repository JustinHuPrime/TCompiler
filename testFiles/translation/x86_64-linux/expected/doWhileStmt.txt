testFiles/translation/x86_64-linux/input/doWhileStmt.tc:
TEXT(_T3foo3bar,
  BLOCK(5,
    MOVE(TEMP(temp6, 8, 8, GP), REG(rdi, 8)),
    JUMP(LABEL(.L3)),
  ),
  BLOCK(3,
    JUMP(LABEL(.L8)),
  ),
  BLOCK(8,
    UNSIGNED2FLOATING(TEMP(temp9, 4, 4, FP), CONSTANT(1, BYTE(1))),
    MOVE(TEMP(temp10, 4, 4, FP), TEMP(temp9, 4, 4, FP)),
    JUMP(LABEL(.L7)),
  ),
  BLOCK(13,
    JUMP(LABEL(.L14)),
  ),
  BLOCK(14,
    MOVE(TEMP(temp15, 4, 4, FP), TEMP(temp10, 4, 4, FP)),
    UNSIGNED2FLOATING(TEMP(temp16, 4, 4, FP), CONSTANT(1, BYTE(2))),
    FMUL(TEMP(temp17, 4, 4, FP), TEMP(temp15, 4, 4, FP), TEMP(temp16, 4, 4, FP)),
    MOVE(TEMP(temp10, 4, 4, FP), TEMP(temp17, 4, 4, FP)),
    JUMP(LABEL(.L12)),
  ),
  BLOCK(19,
    MOVE(TEMP(temp20, 8, 8, GP), TEMP(temp6, 8, 8, GP)),
    SUB(TEMP(temp21, 8, 8, GP), TEMP(temp20, 8, 8, GP), CONSTANT(8, LONG(1))),
    MOVE(TEMP(temp6, 8, 8, GP), TEMP(temp21, 8, 8, GP)),
    JUMP(LABEL(.L18)),
  ),
  BLOCK(18,
    JUMP(LABEL(.L22)),
  ),
  BLOCK(22,
    ZX(TEMP(temp23, 8, 8, GP), CONSTANT(1, BYTE(0))),
    JA(LABEL(.L7), LABEL(.L11), TEMP(temp21, 8, 8, GP), TEMP(temp23, 8, 8, GP)),
  ),
  BLOCK(11,
    MOVE(TEMP(temp25, 4, 4, FP), TEMP(temp10, 4, 4, FP)),
    JUMP(LABEL(.L24)),
  ),
  BLOCK(24,
    MOVE(TEMP(temp2, 4, 4, FP), TEMP(temp25, 4, 4, FP)),
    JUMP(LABEL(.L4)),
  ),
  BLOCK(4,
    MOVE(REG(xmm0, 4), TEMP(temp2, 4, 4, FP)),
    RETURN(),
  ),
)
